plugins/salesforce/src/org/pentaho/di/ui/trans/steps/salesforcedelete/SalesforceDeleteDialog.java

```diff
@@ -548,9 +549,11 @@ private void test() {
         return;
       }
 
+      String realURL = transMeta.environmentSubstitute( meta.getTargetURL() );
+      realUsername = transMeta.environmentSubstitute( meta.getUserName() );
+      String realPassword = transMeta.environmentSubstitute( meta.getPassword() );
```

Comment:
For this line, you need to use Encr.decryptPasswordOptionallyEncrypted( transMeta.environmentSubstitute( xxxx ) ) . See PDI-15652.